### PR TITLE
fall through waterfall for static content

### PR DESCRIPTION
**Pull request by jrperritt**

### CODE REVIEW

**Comment by jrperritt** [comment]
hmm, somehow I'm missing large parts of the original. Looking into it...

**Comment by jrperritt** [comment]
Oh, I see. I'm out of date. There's a `postprocess` function call where there used to be a `layout` function call.

**Comment by jrperritt** [comment]
@smashwilson how does this look? Ideally, we'd return the static content as soon as we get it, but to do that I think we'd have to either return an error to break the waterfall or pass around the `res` object.

File: src/content.js [hunk 1]
```diff
@@ -63,17 +63,20 @@ function mapping(presented, callback) {
...
 // associated attributes at this content ID.
-function content(content_id, callback) {
-  var content_url = urljoin(config.content_service_url(), 'content', encodeURIComponent(content_id));
+function content(content_obj, callback) {
+  var content_url;
+  if (content_obj["proxy-to"]) {
+    content_url = content_obj["proxy-to"];
+  } else {
+    content_url = urljoin(config.content_service_url(), 'content', encodeURIComponent(content_obj["content-id"]));
+  }
```
Comment (smashwilson) [inline src/content.js:86]: I'd personally put an `else if (content_obj["content-id"])` clause here, and add a final `else` that emits an Error to the callback, just to be safe.

**Comment by smashwilson** [comment]
> @smashwilson how does this look? Ideally, we'd return the static content as soon as we get it, but to do that I think we'd have to either return an error to break the waterfall or pass around the res object.

This is pretty reasonable as-is. Like I commented, I think I'd probably try not to assume that a `content_obj` without a `"proxy-to"` key was valid and instead explicitly check for a `"content-id"` too in the other branches; that's just to give us better error messages if the mapping service does something unexpected, though.

Passing around the response object from the content request and piping it directly for the proxy-to case might be better but I'm fine starting with this :+1:

Can you add a test for the proxy-to case?

File: src/content.js (hunk 1 above)
Comment (jrperritt) [inline src/content.js:86]: :+1:

**Comment by jrperritt** [comment]
Cool, I added a unit test as well.

File: test/content.js [hunk 2]
```diff
@@ -48,6 +49,21 @@ describe("/*", function () {
       .expect("Rendered the page content with a layout", done);
   });
 
+  it("handles static content", function (done) {
+    var mapping = nock("http://mapping")
+    .get("/at/https%3A%2F%2Fdeconst.horse%2Ffoo%2Fbar%2Fstatic")
+    .reply(200, { "proxy-to": "https://deconst.horse/static" });
```
Comment (smashwilson) [inline test/content.js:55]: I will never stop laughing at this domain.

:horse:

**Comment by smashwilson** [comment]
:confetti_ball:

**Comment by jrperritt** [comment]
Closes deconst/deconst-docs#63

**Pull request merged**